proposals/interface-types/working-notes/scenarios/points.md

```diff
@@ -0,0 +1,232 @@
+# Paint a vector of points
+
+In this example we look at how sequences are passed into an API. The signature
+of `vectorPaint` is assumed to be:
+
+```
+point ::= pt(i32,i32)
+vectorPaint:(array<point> pts) => returnCode
+```
+
+## Export
+
+It is assumed that the implementation of `vectorPaint_` requires a
+memory-allocated array; each entry of which consists of two contiguous `i32`
+values.
+
+The array is modeled as two values: a pointer to its base and the number of
+elements in the array.
+
+The primary pattern with processing arrays is the for-loop pattern, together
+with array indexing; which is modeled by the `array-to-memory` instruction;
+which iterates a code fragment over a range of numbers.
+
+```
+(@interface datatype @point
+   (tuple i32 i32))
+
+(memory (export "memx") 1)
+
+(func $vectorPaint_ (export ("" "vectorPaint_"))
+  (param i32 i32) (result i32)
+
+(@interface func (export "vectorPaint")
+  (param $pts (array (type $point)))
+  (result (type $returnCode))
+  
+  local.get $pts
+  array.count
+  i32.const #8
+  i32.mul
+  call "malloc"
+  deferred (i32)
+    call "free"
+  end
+  let (local $ptr i32) (result (type $returnCode))
+    local.get $pts
+    array-to-memory #8
+      let (local $pt i32) (local $ptr i32) (local $ix i32)
```

Comment:
Should this be `(local $pt (type $point))`?

Reply:
Yes, you are right. Note that this is subject to change,